File: app/src/main/java/com/fos/fosmvp/base/BaseResponse.java
package com.fos.fosmvp.base;

import java.io.Serializable;

/**
 * 封装服务器返回数据
 *
 */
public class BaseResponse<T>  implements Serializable {
    public String code="-1";
    public String msg;
    public T data;

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getMsg() {
        return msg;
    }

    public void setMsg(String msg) {
        this.msg = msg;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    /**
     * 判断返回值是否成功
     * @return
     */
    public boolean isSucceed(){
        if(code.equals("0"))
            return true;
        else
            return false;
    }
}

File: app/src/main/java/com/fos/fosmvp/common/base/BaseResponse.kt
package com.fos.fosmvp.common.base

import java.io.Serializable

/**
 * 封装服务器返回数据
 *
 */
class BaseResponse<T> : Serializable {
    var code = "-1"
    var msg: String? = null
    var data: T? = null

    /**
     * 判断返回值是否成功
     * @return
     */
    val isSucceed: Boolean
        get() = if (code == "0")
            true
        else
            false
}
